screens/generate_pokemon/dropdown.lua

```diff
@@ -1,135 +1,159 @@
-local gooey = require "screens.generate_pokemon.gooey.gooey"
+local gooey = require "gooey.gooey"
 local gooey_buttons = require "utils.gooey_buttons"
+local gooey_scrolling_list = require "utils.gooey_scrolling_list"
 
 local M = {}
 
 local active = {active=false}
 
-local function button_click(name, button_id, scroll_id)
+local function button_click(data)
```

Comment:
This file was the most changed of everything - I turned dropdown into an object you can "create" and call functions on. All of the data associated with a single dropdown (of which there are several in generate_pokemon) is carried around in this "data" object.